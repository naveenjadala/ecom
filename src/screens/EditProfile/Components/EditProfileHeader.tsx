import {View, Text} from 'react-native';
import React from 'react';
import {Divider} from 'react-native-paper';
import {EditHeaderStyles} from './styles';
import {NavigationProp} from '@react-navigation/native';
import {editProfileStyles} from '../styles';

type EditProfileProps = {
  onCancel: () => void;
  onSave: () => void;
};

/**
 * Renders the EditProfileHeader component.
 *
 * @param {EditProfileProps} props - the properties for the component
 * @return {JSX.Element} the rendered component
 */
const EditProfileHeader = (props: EditProfileProps) => {
  return (
    <>
      <View style={EditHeaderStyles.container}>
        <Text onPress={props.onCancel} style={EditHeaderStyles.text}>
          Cancel
        </Text>
        <Text onPress={props.onSave} style={EditHeaderStyles.text}>
          Save
        </Text>
      </View>
      <Divider />
    </>
  );
};

export default EditProfileHeader;
