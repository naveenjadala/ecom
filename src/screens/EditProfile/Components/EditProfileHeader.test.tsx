import {fireEvent, render, screen} from '../../../test/test-utils';
import EditProfileHeader from './EditProfileHeader';

describe('should render EditProfileHeader', () => {
  test('should render EditProfileHeader', () => {
    render(<EditProfileHeader onCancel={() => {}} onSave={() => {}} />);
    expect(screen).toMatchSnapshot();
  });

  test('should render text element', () => {
    render(<EditProfileHeader onCancel={() => {}} onSave={() => {}} />);
    const getCancelTextView = screen.getByText('Cancel');
    const getSaveTextView = screen.getByText('Save');
    expect(getCancelTextView).toBeTruthy();
    expect(getSaveTextView).toBeTruthy();
  });

  test('should render touchable element cancel', () => {
    const cancel = jest.fn();
    render(<EditProfileHeader onCancel={() => cancel()} onSave={() => {}} />);
    const getClickableElementView = screen.getByText('Cancel');
    fireEvent.press(getClickableElementView);
    expect(cancel).toHaveBeenCalled();
  });

  test('should render touchable element save', () => {
    const save = jest.fn();
    render(<EditProfileHeader onCancel={() => {}} onSave={() => save()} />);
    const getClickableElementView = screen.getByText('Save');
    fireEvent.press(getClickableElementView);
    expect(save).toHaveBeenCalled();
  });
});
