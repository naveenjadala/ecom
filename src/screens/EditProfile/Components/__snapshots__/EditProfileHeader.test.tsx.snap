// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditProfileHeader should render EditProfileHeader 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "minHeight": "100%",
      "minWidth": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "elevation": 2,
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "padding": 15,
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        {
          "color": "#000000",
          "fontSize": 8,
        }
      }
    >
      Cancel
    </Text>
    <Text
      onPress={[Function]}
      style={
        {
          "color": "#000000",
          "fontSize": 8,
        }
      }
    >
      Save
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(202, 196, 208, 1)",
          "height": 0.5,
        },
        undefined,
        false,
        false,
        undefined,
      ]
    }
  />
</RCTSafeAreaView>
`;
